File: src/api/cache_namespace.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct CacheNamespace<'a> {
    cfg: &'a Configuration,
}

impl<'a> CacheNamespace<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Create cache namespace
    pub async fn create(
        &self,
        cache_create_namespace_in: CacheCreateNamespaceIn,
    ) -> Result<CacheCreateNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.cache.namespace.create")
            .with_body(cache_create_namespace_in)
            .execute(self.cfg)
            .await
    }

    /// Get cache namespace
    pub async fn get(
        &self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> Result<CacheGetNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.cache.namespace.get")
            .with_body(cache_get_namespace_in)
            .execute(self.cfg)
            .await
    }
}

File: coyote/apis/cache_namespace.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    CacheCreateNamespaceIn,
    CacheCreateNamespaceOut,
    CacheGetNamespaceIn,
    CacheGetNamespaceOut,
)


class CacheNamespaceAsync(ApiBase):
    async def create(
        self,
        cache_create_namespace_in: CacheCreateNamespaceIn,
    ) -> CacheCreateNamespaceOut:
        """Create cache namespace"""
        body = cache_create_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.cache.namespace.create",
            body=body,
            response_type=CacheCreateNamespaceOut,
        )

    async def get(
        self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> CacheGetNamespaceOut:
        """Get cache namespace"""
        body = cache_get_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.cache.namespace.get",
            body=body,
            response_type=CacheGetNamespaceOut,
        )


class CacheNamespace(ApiBase):
    def create(
        self,
        cache_create_namespace_in: CacheCreateNamespaceIn,
    ) -> CacheCreateNamespaceOut:
        """Create cache namespace"""
        body = cache_create_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.cache.namespace.create",
            body=body,
            response_type=CacheCreateNamespaceOut,
        )

    def get(
        self,
        cache_get_namespace_in: CacheGetNamespaceIn,
    ) -> CacheGetNamespaceOut:
        """Get cache namespace"""
        body = cache_get_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.cache.namespace.get",
            body=body,
            response_type=CacheGetNamespaceOut,
        )
